File: erlang/rna-transcription/rna_transcription.erl
-module(rna_transcription).

-export([to_rna/1]).

%to_rna(Gene) -> Output.

to_rna("C") -> "G";
to_rna("G") -> "C";
to_rna("A") -> "U";
to_rna("T") -> "A";
to_rna([Head|Tails]) -> to_rna([Head]) ++ to_rna(Tails);
to_rna(_) -> [].

File: elixir/rna-transcription/rna_transcription.exs
defmodule RNATranscription do
  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

  iex> RNATranscription.to_rna('ACTG')
  'UGAC'
  """
  @spec to_rna([char]) :: [char]
  def to_rna(dna = [ nuct | strand ]) do
    rna(nuct) ++ to_rna(strand)
  end
  def to_rna([]), do: []

  def rna(?A), do: 'U'
  def rna(?T), do: 'A'
  def rna(?G), do: 'C'
  def rna(?C), do: 'G'

end
